WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -178,9 +195,15 @@ final class ProductSelectorViewModel: ObservableObject {
         guard let selectedProduct = products.first(where: { $0.productID == productID }) else {
             return
         }
-        if let onProductSelected = onProductSelected {
+
+        if supportsMultipleSelections, let onProductSelected {
```

Comment:
I think we can simplify these if-else blocks, I let you find a more optimal and shorter way to do it :)

Reply:
Right! I refactored these using `guard` on https://github.com/woocommerce/woocommerce-ios/pull/8935/commits/47c84cac6403d261a1c6818807681adc06f0abde . Works as expected and all test pass, but happy to get a second look to the changes, as personally I still don't find them more readable using guard rather than "if else", I have to get better acquainted :P